test/cmd/apply.sh

```diff
@@ -232,18 +232,18 @@ __EOF__
   kube::test::get_object_assert pods "{{range.items}}{{${id_field:?}}}:{{end}}" ''
   kubectl delete pvc b-pvc 2>&1 "${kube_flags[@]:?}"
 
-  ## kubectl apply --prune --prune-whitelist
+  ## kubectl apply --prune --prune-allowlist
   # Pre-Condition: no POD exists
   kube::test::get_object_assert pods "{{range.items}}{{${id_field:?}}}:{{end}}" ''
   # apply pod a
   kubectl apply --prune -l prune-group=true -f hack/testdata/prune/a.yaml "${kube_flags[@]:?}"
   # check right pod exists
   kube::test::get_object_assert 'pods a' "{{${id_field:?}}}" 'a'
-  # apply svc and don't prune pod a by overwriting whitelist
-  kubectl apply --prune -l prune-group=true -f hack/testdata/prune/svc.yaml --prune-whitelist core/v1/Service 2>&1 "${kube_flags[@]:?}"
+  # apply svc and don't prune pod a by overwriting allowlist
+  kubectl apply --prune -l prune-group=true -f hack/testdata/prune/svc.yaml --prune-allowlist core/v1/Service 2>&1 "${kube_flags[@]:?}"
```

Comment:
this represents a breaking change to the command line API. if we want to double-register the flag, that could be ok.

Reply:
also, this flag wasn't actually changed in staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

Reply:
It must already be double registered, or this change would break the test. I don't see any kubectl flag changes in this PR.

Reply:
this change is breaking the test... this is the cause of the test-integration failure:

```
Error: unknown flag: --prune-allowlist
```

Reply:
so the flag would also need to be changed within apply.go? or would you recommend reverting this change?

Reply:
I would definitely recommend changing things with compatibility implications in a separate PR. Revert it here.

Reply:
will push this shortly, thank you!